sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs

```diff
@@ -0,0 +1,123 @@
+        /// overriding <see href="https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/samples/Pipeline.md">transport</see>,
+        /// enabling <see href="https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/samples/Diagnostics.md">diagnostics</see>,
+        /// and controlling <see href="https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/samples/Configuration.md">retry strategy</see>.
+        /// </para>
+        /// </remarks>
+        public TimeSeriesInsightsClient(string environmentFqdn, TokenCredential credential, TimeSeriesInsightsClientOptions options)
+        {
+            Argument.AssertNotNull(options, nameof(options));
+
+            _clientDiagnostics = new ClientDiagnostics(options);
+
+            options.AddPolicy(new BearerTokenAuthenticationPolicy(credential, GetAuthorizationScopes()), HttpPipelinePosition.PerCall);
+            _httpPipeline = HttpPipelineBuilder.Build(options);
+
+            string versionString = options.GetVersionString();
+            _modelSettingsRestClient = new ModelSettingsRestClient(_clientDiagnostics, _httpPipeline, environmentFqdn, versionString);
+            _queryRestClient = new QueryRestClient(_clientDiagnostics, _httpPipeline, environmentFqdn, versionString);
+            _timeSeriesHierarchiesRestClient = new TimeSeriesHierarchiesRestClient(_clientDiagnostics, _httpPipeline, environmentFqdn, versionString);
+            _timeSeriesTypesRestClient = new TimeSeriesTypesRestClient(_clientDiagnostics, _httpPipeline, environmentFqdn, versionString);
+        }
+
+        /// <summary>
+        /// Creates a new instance of the <see cref="TimeSeriesInsightsClient"/> class, provided for unit testing purposes only.
+        /// </summary>
+        protected TimeSeriesInsightsClient()
+        {
+        }
+
+        /// <summary>
+        /// Returns the model settings which includes model display name, Time Series ID properties and default type ID. Every Gen2 environment has a model that is automatically created.
+        /// </summary>
+        /// <param name="clientSessionId"> Optional client session ID. Service records this value. Allows the service to trace a group of related operations across services, and allows the customer to contact support regarding a particular group of requests. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <returns>The deserialized application/json model settings digital twin and the http response <see cref="Response{ModelSettingsResponse}"/>.</returns>
+        /// <example>
+        /// This sample demonstrates getting and deserializing a model settings.
+        ///
+        /// <code snippet="Snippet:TimeSeriesInsightsGetModelSettings">
+        /// </code>
+        /// </example>
+        public virtual Task<Response<ModelSettingsResponse>> GetAsync(string clientSessionId = null, CancellationToken cancellationToken = default)
```

Comment:
Are we also designing the APIs or are these basically placeholders ? 
if it's the former, `GetAsync` what? are we doing API design reviews before implementation?

Reply:
I think it's the latter though

Reply:
This API is just a place holder in order to demo for the TSI team authentication and running the tests.